### DISCUSSION TITLE
hm-400 no power to the grid

### DISCUSSION THREAD

**Discussion by kthemall** [tbnobody/OpenDTU#996] [Q&A]
### What happened?

the hoymiles hm-400 does not deliver power to the grid. maybe the logs tell someone what the problem is?? 
systeminfo:

Hostname | OpenDTU1
-- | --
v4.4.4
0.1.25
v23.6.1
generic
Aktuell!
Software reset CPU
Software reset CPU
42
0 days 00:04:39

thx tom

### To Reproduce Bug

does not work either.

### Expected Behavior

if i send a power command it should raise or degrade the power of the inverter correctly.

### Install Method

Pre-Compiled binary from GitHub

### What git-hash/version of OpenDTU?

v23.6.1

### Relevant log/trace output

```Shell
Websocket: [/livedata][2] disconnect
11:44:15.090 > Fetch inverter: 112183814498
11:44:15.154 > TX RealTimeRunData Channel: 40 --> 15 83 81 44 98 80 11 20 52 80 0B 00 64 79 B9 EF 00 00 00 00 00 00 00 00 D4 0B 37 
11:44:15.207 > Interrupt received
11:44:15.276 > RX Channel: 3 --> 95 83 81 44 98 83 81 44 98 01 00 01 01 07 00 29 00 6B 00 00 A9 0C 0F 3E 09 23 6F 
11:44:15.655 > RX Period End
11:44:15.655 > Last missing
11:44:15.655 > Request retransmit: 2
11:44:15.655 > TX RequestFrame Channel: 61 --> 15 83 81 44 98 80 11 20 52 82 AA 
11:44:15.972 > Interrupt received
11:44:16.021 > RX Channel: 23 --> 95 83 81 44 98 83 81 44 98 82 13 8A 00 00 00 00 00 00 00 00 00 F9 00 06 5D 1E 32 
11:44:16.071 > RX Period End
11:44:16.071 > Success
11:44:16.126 > TX AlarmData Channel: 75 --> 15 83 81 44 98 80 11 20 52 80 11 00 64 79 B9 EF 00 00 00 00 00 00 00 00 0E 10 EC 
11:44:16.206 > Interrupt received
11:44:16.263 > RX Channel: 3 --> 95 83 81 44 98 83 81 44 98 84 ED 43 40 94 00 06 43 F6 86 22 0D 8D 00 5A 77 5C 8D 
11:44:16.591 > RX Period End
11:44:16.591 > Middle missing
11:44:16.591 > Request retransmit: 1
11:44:16.591 > TX RequestFrame Channel: 3 --> 15 83 81 44 98 80 11 20 52 81 A9 
11:44:16.714 > RX Period End
11:44:16.715 > Middle missing
11:44:16.715 > Request retransmit: 1
11:44:16.715 > TX RequestFrame Channel: 23 --> 15 83 81 44 98 80 11 20 52 81 A9 
11:44:16.821 > RX Period End
11:44:16.821 > Middle missing
11:44:16.821 > Request retransmit: 1
11:44:16.821 > TX RequestFrame Channel: 40 --> 15 83 81 44 98 80 11 20 52 81 A9 
11:44:16.873 > Interrupt received
11:44:16.941 > RX Channel: 75 --> 95 83 81 44 98 83 81 44 98 01 00 01 80 01 00 01 88 AE 88 AE 00 00 00 00 B0 0C A9 
11:44:17.011 > RX Period End
11:44:17.011 > Middle missing
11:44:17.011 > Request retransmit: 2
11:44:17.011 > TX RequestFrame Channel: 61 --> 15 83 81 44 98 80 11 20 52 82 AA 
11:44:17.077 > Interrupt received
11:44:17.200 > RX Channel: 75 --> 95 83 81 44 98 83 81 44 98 02 00 02 37 A4 37 A4 0A A3 F9 B0 B0 2E 00 03 72 AD 37 
11:44:17.256 > RX Period End
11:44:17.256 > Middle missing
11:44:17.256 > Request retransmit: 3
11:44:17.256 > TX RequestFrame Channel: 75 --> 15 83 81 44 98 80 11 20 52 83 AB 
11:44:17.315 > Interrupt received
11:44:17.511 > RX Channel: 23 --> 95 83 81 44 98 83 81 44 98 03 72 AD 0A AC 01 BE 80 24 00 04 43 F5 43 F5 07 F9 0E 
11:44:17.574 > RX Period End
11:44:17.574 > Success
```

### Anything else?

no thanks

**Comment by DPO99** [comment]
Please share a screenshot from the live data view.

**Comment by kthemall** [comment]
![opendtu](https://github.com/tbnobody/OpenDTU/assets/78484370/eb419303-1e38-480b-93b9-78a065e26c5b)

**Comment by tbnobody** [comment]
Only 1,3W on String 1 seems to be much to low to start producing....

**Comment by DPO99** [comment]
Same here if the inverter is turned off. 

![image](https://github.com/tbnobody/OpenDTU/assets/125746530/eb8d5b12-98c3-4f05-bbf6-dbdaaff910e1)

Can you try to turn it on?

![image](https://github.com/tbnobody/OpenDTU/assets/125746530/f10a9d5c-301f-4c10-bdf9-56fd0243aefc)

![image](https://github.com/tbnobody/OpenDTU/assets/125746530/e9cccf4e-b24f-4232-9670-091415b632ec)

**Comment by tbnobody** [comment]
If you show us a image of the event log we would see if the inverter is turned off

**Comment by kthemall** [comment]
![IMG_2348](https://github.com/tbnobody/OpenDTU/assets/78484370/6dface88-f7c7-458b-ac79-fb2ccfdde410)
![IMG_2349](https://github.com/tbnobody/OpenDTU/assets/78484370/914ce4b7-9e5a-49ad-b353-eb73604cccdb)

hi, hope you can see something strange in the pics. thx tom

**Comment by kthemall** [comment]
sorry forgot to mention, i use the inverter on a 8s lifepo battery.
after a restart of the inverter (via opendtu) it delivers power as it should. but just for a few minutes, then i have again the yellow screen.
maybe it has to do with time settings? isn´t it possible to deliver power to the grid after a certain time?

**Comment by tictrick** [comment]
How big is the battery?

Can it be that the voltage of the battery drops and the BMS switches off as a result? The inverter may then first need a reboot so that it can produce again.

**Reply by kthemall** [reply]
hi, the battery has 280ah and if it goes under 10% i turn off the AC-Side of the hm-400 automatically (not dc, that is always connected and has enough power). if it goes up to 50% i turn it on automatically. problem? thx tom

**Reply by DPO99** [reply]
The capacity ist not relevant as Well. Which Batterie Type? Voltage? Etc.

Why do you dont Turn Off the HM? Opendtu ist able To handle this via mqtt or API.

**Comment by tbnobody** [comment]
I would convert this to a discussion since, as of now, it is not an OpenDTU issue.

**Reply by kthemall** [reply]
ok, thx tom

**Comment by kthemall** [comment]
eve lifepo4 8s between 26,5v and 28,5v always on Gesendet mit der GMX iPad AppAm 03.06.23 um 13:41 schrieb DPO99
                
            
            
                
                    Von: "DPO99" ***@***.***>Datum: 3. Juni 2023An: "tbnobody/OpenDTU" ***@***.***>Cc: "kthemall" ***@***.***>,"Author" ***@***.***>Betreff: Re: [tbnobody/OpenDTU] hm-400 no power to the grid (Discussion #996)
                
                
The capacity ist not relevant as Well. Eh ich Batterie Type? Voltage? Etc.

**Comment by jaybee79** [comment]
Only an idea: with a voltage that low there has to be a high current to reach the 400W. Maybe some internal protection of the HM400 doesn't alow running at that high current und turns off the inverter. With lets say 26V (some loss on the cable) you would need more than 15,4A (400W/26V=15,38A) at the input. There is an absolute maximum of 15A in the datasheet for short circuit current and a 12,5A limit for "normal" input current. Could you try to set the power limit to lets say 75%/300W? With that the input power should stay around 12A . I'm running some HM400/800 on batteries too but never had that issues because I'm using 48V battery systems (one lead acid, one on LiFePO4).

**Reply by kthemall** [reply]
hi, the 400w will never be reached, but that is not the problem, i´m ok with about 230W. it never goes above 10,5A. i think it has to do if i disconnect and reconnect AC-Power. maybe it has to do with the python script i am using to set the Watt. today it started without problems. went from yellow background to blue and started delivering energy to the grid. 
hmmm

**Reply by DPO99** [reply]
Why do you disconnect the AC? This is not the expected situation for the HM. I never disconnect AC/DC and got no problems with the HM.

**Comment by kthemall** [comment]
oh, i do it to protect my lifepo battery as explained.and why not expected? on a grid disconnect activated by power provider AC also is disconnected 😂😂😂….hopefullyGesendet mit der GMX iPad AppAm 05.06.23 um 20:12 schrieb DPO99
                
            
            
                
                    Von: "DPO99" ***@***.***>Datum: 5. Juni 2023An: "tbnobody/OpenDTU" ***@***.***>Cc: "Author" ***@***.***>,"kthemall" ***@***.***>Betreff: Re: [tbnobody/OpenDTU] hm-400 no power to the grid (Discussion #996)
                
                
Why do you disconnect the AC? This is not the expected situation for the HM. I never disconnect AC/DC and got no problems with the HM.
